src/main/java/edu/hawaii/its/api/service/GroupPathService.java

```diff
@@ -112,6 +112,16 @@ private Group getGroup(String groupPath) {
         return grouperApiService.findGroupsResults(groupPath).getGroup();
     }
 
+    public List<String> getBasePathAndSubPathFromSubPath(String groupPath) {
+        List<String> groupPaths = new ArrayList<>();
+        if (isOwnersGroupPath(groupPath) || isBasisGroupPath(groupPath) || isIncludeGroupPath(groupPath)
+                || isExcludeGroupPath(groupPath)) {
+            groupPaths.add(getGroupingPath(groupPath));
+        }
+        groupPaths.add(groupPath);
+        return groupPaths;
+    }
```

Comment:
You might be able to use the getExtension and isGroupExtension functions to make this even nicer. For the name of the function, maybe something like extractPaths or getGroupingGroupPaths could work.